fedimint-server/src/consensus/aleph_bft/finalization_handler.rs

```diff
@@ -1,22 +1,37 @@
+use aleph_bft::{NodeIndex, Round};
 use fedimint_core::PeerId;
 
 use super::data_provider::UnitData;
 
+pub struct OrderedUnit {
+    pub creator: PeerId,
+    pub round: Round,
+    pub data: Option<UnitData>,
+}
+
 pub struct FinalizationHandler {
-    sender: async_channel::Sender<(UnitData, PeerId)>,
+    sender: async_channel::Sender<OrderedUnit>,
 }
 
 impl FinalizationHandler {
-    pub fn new(sender: async_channel::Sender<(UnitData, PeerId)>) -> Self {
+    pub fn new(sender: async_channel::Sender<OrderedUnit>) -> Self {
         Self { sender }
     }
 }
 
 impl aleph_bft::FinalizationHandler<UnitData> for FinalizationHandler {
-    fn data_finalized(&mut self, unit_data: UnitData, creator: aleph_bft::NodeIndex) {
+    fn data_finalized(&mut self, _data: UnitData) {
+        unreachable!("This method is not called")
+    }
+
+    fn unit_finalized(&mut self, creator: NodeIndex, round: Round, data: Option<UnitData>) {
         // the channel is unbounded
         self.sender
-            .try_send((unit_data, super::to_peer_id(creator)))
+            .try_send(OrderedUnit {
+                creator: super::to_peer_id(creator),
+                round,
+                data,
+            })
             .ok();
```

Comment:
note: We need to use the less stable interface here so we can learn the creator id.